core/loadpoint.go

```diff
@@ -928,10 +928,19 @@ func (lp *Loadpoint) scalePhases(phases int) error {
 	}
 
 	if lp.GetPhases() != phases {
-		// disable charger - this will also stop the car charging using the api if available
-		if err := lp.setLimit(0, true); err != nil {
-			return err
+
+		// set new current
+
+		current := lp.chargeCurrent * float64(lp.GetPhases())
+		if phases > 1 {
+			current = current / float64(lp.maxActivePhases())
 		}
+		if current < lp.GetMinCurrent() {
+			current = lp.GetMinCurrent()
+		} else if current > lp.GetMaxCurrent() {
+			current = lp.GetMaxCurrent()
+		}
+		_ = lp.setLimit(current, true)
```

Comment:
```suggestion
		err := lp.setLimit(lp.GetMinCurrent(), true); if err != nil { ...
		}
```

Reply:
An der Stelle macht Fehlerhandling imho keinen Sinn. Die Änderung des Ladestroms vermeidet nur Leistungsabweichungen nach unten und oben. Schlägt diese fehl geht die Ladeleistung zwar unmittelbar nach der Umschaltung hoch oder runter, dann greift aber die Regelung und regelt den Sollwert ein.

Reply:
Fehlerhandling macht immer Sinn. Wenn ich noch nicht mit dem Charger reden kann muss ich auch nicht weiter auf ihn einreden ;)

Reply:
Sehe ich in der Absolutheit nicht so. Fehlerbehandlung ist essentiell wo der Erfolg eines Schritts für den Erfolg des nächsten Schritts notwendig ist. Der Erfolg dieses Aufrufs ist für den weiteren Ablauf nicht notwendig. Steht der Charger dauerhaft nicht zur Verfügung kommt der Fehler dann bei der Phasenumschaltung. Ist es ein vorübergehendes Problem kann man das Fehlschlagen der Änderung des Ladestroms ignorieren.  Siehe Zeile 609 - da ist der Präzedenzfall :-)